tests/collections/tts/test_spectrogram_enhancer.py

```diff
@@ -0,0 +1,186 @@
+
+
+@pytest.fixture
+def enhancer_with_fastpitch(enhancer_config_with_fastpitch):
+    return SpectrogramEnhancerModel(cfg=enhancer_config_with_fastpitch.model)
+
+
+@pytest.fixture
+def sample_input(batch_size=15, max_length=1000):
+    generator = torch.Generator()
+    generator.manual_seed(0)
+
+    lengths = torch.randint(max_length // 4, max_length - 7, (batch_size,), generator=generator)
+    condition = torch.randn((batch_size, 80, 1000), generator=generator)
+
+    return condition, lengths
+
+
+@pytest.mark.unit
+def test_pad_spectrogram(enhancer, sample_input):
+    condition, lengths = sample_input
+    output = enhancer.pad_spectrogram(condition)
+
+    assert output.size(-1) >= condition.size(-1)
+
+
+@pytest.mark.unit
+def test_generator_pass_keeps_size(enhancer: SpectrogramEnhancerModel, sample_input):
+    condition, lengths = sample_input
+    output = enhancer.forward(condition=condition, lengths=lengths)
+
+    assert output.shape == condition.shape
+
+
+@pytest.mark.unit
+def test_discriminator_pass(enhancer: SpectrogramEnhancerModel, sample_input):
+    condition, lengths = sample_input
+    condition = rearrange(condition, "b c l -> b 1 c l")
+    logits = enhancer.D(x=condition, condition=condition, lengths=lengths)
+
+    assert logits.shape == lengths.shape
+
+
+@pytest.mark.unit
+def test_nemo_save_load(enhancer: SpectrogramEnhancerModel, tmp_path):
+    path = tmp_path / "test-enhancer-save-load.nemo"
+
+    enhancer.save_to(path)
+    return SpectrogramEnhancerModel.restore_from(path)
+
+
+@pytest.mark.with_downloads
+@pytest.mark.unit
+def test_nemo_save_load_enhancer_with_fastpitch(enhancer_with_fastpitch: SpectrogramEnhancerModel, tmp_path):
+    path = tmp_path / "test-enhancer-save-load.nemo"
+
+    enhancer_with_fastpitch.save_to(path)
+    restored_enhancer = SpectrogramEnhancerModel.restore_from(path)
+    assert restored_enhancer.spectrogram_model is None
+
+
+# @pytest.mark.with_downloads
+# @pytest.mark.unit
+def fastpitch_loads_data_inside_enhancer(enhancer_with_fastpitch: SpectrogramEnhancerModel):
+    assert len(enhancer_with_fastpitch._train_dl.dataset) > 0
+
+
+# @pytest.mark.with_downloads
+# @pytest.mark.unit
+def enhancer_with_fastpitch_training_step_discriminator(enhancer_with_fastpitch: SpectrogramEnhancerModel):
+    batch = next(iter(enhancer_with_fastpitch._train_dl))
+    enhancer_with_fastpitch.training_step(batch, batch_idx=0, optimizer_idx=0)
+
+
+# @pytest.mark.with_downloads
+# @pytest.mark.unit
+def enhancer_with_fastpitch_training_step_generator(enhancer_with_fastpitch: SpectrogramEnhancerModel):
+    batch = next(iter(enhancer_with_fastpitch._train_dl))
+    enhancer_with_fastpitch.training_step(batch, batch_idx=0, optimizer_idx=1)
+
+
+@pytest.mark.with_downloads
+@pytest.mark.unit
+def test_enhancer_with_fastpitch(enhancer_with_fastpitch: SpectrogramEnhancerModel):
+    # long setup for enhancer_with_fastpitch, around 1 min
```

Comment:
Should this be a unit test if it takes 1m+ to set up and run?

Reply:
Good question! Do we have a working way to say THIS TAKES LONG in our testing setup? The `.unit` mark here is just copy-paste

Reply:
@rlangman please take a look at how tests are made, thanks

Reply:
Ah, I found this in the development docs:
> Unit tests must be fast: <1s ideally, <10s at worst, and >10s needs to have an exception from NeMo Core team.

So I guess this should be an integration test ("These tests are longer running and are used to confirm a behavior.").

Reply:
You should be able to mark them with `@pytest.mark.integration`, but I only see a few Hydra tests that use it in NeMo. @okuchaiev can you advise on best practice here? Should we be using this marker more, or is something else more appropriate?

Reply:
Removed the tests because the configuration they used was also removed